src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -493,5 +653,261 @@ public async Task<ContentItem> CloneAsync(ContentItem contentItem)
+                {
+                    updatedVersion.ModifiedUtc = modifiedUtc;
+                }
+                if (publishedUtc.HasValue)
+                {
+                    updatedVersion.PublishedUtc = publishedUtc;
+                }
+            }
+
+            // TODO implement sessionhelper.Save
+            _session.Save(updatingVersion);
+
+            return result;
+        }
+
+        private async Task RemoveLatestVersionAsync(ContentItem contentItem, IEnumerable<ContentItem> evictionVersions)
+        {
+            ContentItem latestVersion;
+            if (evictionVersions == null)
+            {
+                latestVersion = await _session.Query<ContentItem, ContentItemIndex>()
+                    .Where(x => x.ContentItemId == contentItem.ContentItemId && x.Latest)
+                    .FirstOrDefaultAsync();
+            }
+            else
+            {
+                latestVersion = evictionVersions.FirstOrDefault(x => x.Latest);
+            }
+
+            if (latestVersion != null)
+            {
+                var publishedVersion = evictionVersions.FirstOrDefault(x => x.Published);
+
+                var removeContext = new RemoveContentContext(contentItem, publishedVersion == null);
+
+                await Handlers.InvokeAsync((handler, context) => handler.RemovingAsync(context), removeContext, _logger);
+                latestVersion.Latest = false;
+                _session.Save(latestVersion);
+                await ReversedHandlers.InvokeAsync((handler, context) => handler.RemovedAsync(context), removeContext, _logger);
+            }
+        }
+
+        private async Task RemovePublishedVersionAsync(ContentItem contentItem, IEnumerable<ContentItem> evictionVersions)
+        {
+            ContentItem publishedVersion;
+            if (evictionVersions == null)
+            {
+                publishedVersion = await _session.Query<ContentItem, ContentItemIndex>()
+                    .Where(x => x.ContentItemId == contentItem.ContentItemId && x.Published)
+                    .FirstOrDefaultAsync();
+            }
+            else
+            {
+                publishedVersion = evictionVersions.FirstOrDefault(x => x.Published);
+            }
+
+            if (publishedVersion != null)
+            {
+                var removeContext = new RemoveContentContext(contentItem, true);
+
+                await Handlers.InvokeAsync((handler, context) => handler.RemovingAsync(context), removeContext, _logger);
+                publishedVersion.Published = false;
+                _session.Save(publishedVersion);
+                await ReversedHandlers.InvokeAsync((handler, context) => handler.RemovedAsync(context), removeContext, _logger);
+            }
+        }
+
+        private async Task RemoveVersionsAsync(ContentItem contentItem, IEnumerable<ContentItem> evictionVersions)
+        {
+            IEnumerable<ContentItem> activeVersions;
+            if (evictionVersions != null)
```

Comment:
yes, should be `if (evictionVersions == null)` (so done)

It is possible evictionVersions is null (intentionally), because this will get called by either
- ImportAsync (which will do the larger query across a number of items for the evictionVersions), and will not be null, or
- `UpdateContentItemVersionAsync(), - the standalone public api method, which will then call the private helpers with a null eviction versions.

So done this way to reduce the requirement for (multiple) queries and keep the logic in one place.